dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -206,6 +206,26 @@ ColumnDefinesPtr generateStoreColumns(const ColumnDefines & table_columns, bool
     }
     return columns;
 }
+
+void convertStringTypeToDefault(DataTypePtr & type)
+{
+    static const auto default_string_type = DataTypeFactory::instance().getOrSet(DataTypeString::getDefaultName());
+    static const auto default_nullable_string_type
+        = DataTypeFactory::instance().getOrSet(DataTypeString::getNullableDefaultName());
+
+    if (removeNullable(type)->getTypeId() != TypeIndex::String)
+        return;
+    if (type->isNullable())
+        type = default_nullable_string_type;
+    else
+        type = default_string_type;
```

Comment:
```suggestion
    if (removeNullable(type)->getTypeId() != TypeIndex::String)
        return;
    if (type->isNullable())
        type = DataTypeFactory::instance().getOrSet(DataTypeString::getNullableDefaultName());
    else
        type = DataTypeFactory::instance().getOrSet(DataTypeString::getDefaultName());
```

IMO, we can directly use `DataTypeFactory::getOrSet` instead of storing a static type ptr on this method. `DataTypeFactory::getOrSet` itself is lightweight enough.